### ISSUE TITLE
Upgrade fails from APS 2.0-10 to version 2.0-15

### ISSUE THREAD

**Issue by for93t** [SpamExperts/aps2-addon#7]
When trying to upgrade from APS 2.0-10 to version 2.0-15 this error occurs:
```
Task ID
13893964
Task name
Upgrading resources of application 'SpamExperts'
Task description
Upgrading resources of application SpamExperts 2.0-10 to version 2.0-15
Belongs to groups
2699:154194302
Parent task ID
13893963
Queue status
Failed
Start not earlier than
Jan-25-2017 20:03
Method name
taskUpgradeApp2x on SCREF:SaaSManagerTasks:0
Last execution output
Internal error: APSC: Upgrade to ver. 2.0-15 : resource 399ad95e-0941-4e38-9d38-9bd9e02beea7 of type 'service' (http://aps.spamexperts.com/app/service/1.3) for APS application 'SpamExperts-2.0-15' : The operation with name 'onContextAvailable' is not found..
```
Aps support suggested the isue was due to the domain example.com. The domain exists in SE, it shows in Odin as not protected. When trying to unprotect the domain and update the following error is seen:
```
Internal error: APSC: Upgrade to ver. 2.0-15 : resource 399ad95e-0941-4e38-9d38-9bd9e02beea7 of type 'service' (http://aps.spamexperts.com/app/service/1.3) for APS application 'SpamExperts-2.0-15' : Internal Server Error: Unknown method "onContextAvailabeSubscribe"..
```

**Comment by for93t** [comment]
The problem with the `aciel-geomatics.com` domain is that during the provisioning process the application cannot update it's MX records are they are already updated.

**Comment by adaivan** [comment]
Odin provided the following:

I have checked the reported issue and could see that from the error_log in endpoint server it is getting the following aps-runtime-error,
```

+-------------+
[2017-01-26 07:18:03.841288 UTC] TRACE #Request to APS controller [apsc-request-5889a2abcd629]:
POST https://10.2.203.188:6308/aps/2/resources/399ad95e-0941-4e38-9d38-9bd9e02beea7/aps/subscriptions
APS-Controller-URI: https://10.2.203.188:6308/
APS-Request-ID: 9dd9bdc4-4033-4cf0-8326-72be03c644d2
Content-type: application/json
Connection: Keep-Alive
Keep-Alive: 300
Accept: */*
APS-Transaction-ID: 32019-24092
{"event":"http:\/\/aps-standard.org\/core\/events\/available","source":{"type":"http:\/\/aps.spamexperts.com\/app\/context\/2.0"},"handler":"onContextAvailable"}
[2017-01-26 07:18:03.914561 UTC] TRACE #Response from APS controller [apsc-request-5889a2abcd629]:
HTTP/1.1 500 Internal Server Error
Connection: Close
Content-Length: 111
Content-Type: application/json
Date: Thu, 26 Jan 2017 07:18:05 GMT

{
"error": "APS::Util::Exception",
"message": "The operation with name 'onContextAvailable' is not found."
}

aps-runtime-error: {
"code": 500,
"error": "Rest::RestException",
"message": "The operation with name 'onContextAvailable' is not found.",
"details": {
"error": "APS::Util::Exception",
"message": "The operation with name 'onContextAvailable' is not found."
},
+-------------+


From the above, it is getting the function "onContextAvailable" that is present with the package returning the error.

The function associated from the package is ,

+---------+
/**
* @verb(POST)
* @path("/onContextAvailable")
* @param("http://aps-standard.org/types/core/resource/1.0#Notification",body)
*/
public function onContextAvailable($event)
{
$this->logger->info(__METHOD__ . ": start");

/** @var context $context */
$context = $this->APSC()->getResource($event->source->id);
if (method_exists($context, 'autoprotectAll')) {
$context->autoprotectAll();
$this->logger->err("works");
} else {
$this->logger->err(__METHOD__ . ": Resource {$event->source->id} has wrong datatype - "
. get_class($context));
}

$this->logger->info(__METHOD__ . ": stop");
}

+---------+
```

Please consider contacting the packager with the details of the error so that they can further investigate with the issue.

**Comment by for93t** [comment]
I have checked the code and so far I don't have ideas on what can cause the issue that the client has reported.

The "missing operation" has been added in [v2.0-9](https://github.com/SpamExperts/aps2-addon/blame/eff8ee455c257a3aaa5b2ee491147444e3829f2a/src/scripts/service.php#L302-L321). The client uses v2.0.10 so the code must be there. I'm unable to reproduce the issue on our systems.

So I'd suggest to investigate the issue on  the client's server with Odin support.
